src/clr-angular/utils/drag-and-drop/providers/drag-handle-registrar.spec.ts

```diff
@@ -0,0 +1,107 @@
+import {ClrDragHandleRegistrar} from "./drag-handle-registrar";
+
+export default function(): void {
+    describe("Drag Handle Registrar", function() {
+        let fixture: ComponentFixture<any>;
+        let testComponent: DragHandleTestComponent;
+
+        const draggableEl = document.createElement("div");
+        const customHandleEl = document.createElement("button");
+
+        // Providers
+        let dragHandleRegistrar: any;
+        let dragEventListener: any;
+
+        beforeEach(() => {
+            TestBed.configureTestingModule(
+                {declarations: [DragHandleTestComponent], providers: [ClrDragAndDropEventBus]});
+            fixture = TestBed.createComponent(DragHandleTestComponent);
+            testComponent = fixture.componentInstance;
+            dragHandleRegistrar = fixture.debugElement.injector.get(ClrDragHandleRegistrar);
+            dragEventListener = fixture.debugElement.injector.get(ClrDragEventListener);
+        });
+
+        afterEach(() => {
+            fixture.destroy();
+        });
+
+        it("should be able to register element as default handle on assignment", function() {
+            dragHandleRegistrar.defaultHandleEl = draggableEl;
+            expect(dragEventListener.draggableEl).toBe(draggableEl);
+            expect(draggableEl.hasListener).toBeTruthy();
+            expect(draggableEl.classList.contains("drag-handle")).toBeTruthy();
+        });
+
+        it("should be able to register custom element as handle", function() {
+            dragHandleRegistrar.registerCustomHandle(customHandleEl);
+
+            expect(dragHandleRegistrar.customHandle).toBe(customHandleEl);
+            expect(dragEventListener.draggableEl).toBe(customHandleEl);
+
+            expect(customHandleEl.hasListener).toBeTruthy();
+            expect(customHandleEl.classList.contains("drag-handle")).toBeTruthy();
+        });
+
+        it("should be able to register custom element as drag handle after default handle is set", function() {
+            dragHandleRegistrar.defaultHandleEl = draggableEl;
+            dragHandleRegistrar.registerCustomHandle(customHandleEl);
+            // Once custom handle gets registered, listeners and drag styles should be removed from default element.
+            expect(draggableEl.hasListener).toBeFalsy();
+            expect(draggableEl.classList.contains("drag-handle")).toBeFalsy();
+
+            expect(dragHandleRegistrar.customHandle).toBe(customHandleEl);
+            expect(dragEventListener.draggableEl).toBe(customHandleEl);
+
+            expect(customHandleEl.hasListener).toBeTruthy();
+            expect(customHandleEl.classList.contains("drag-handle")).toBeTruthy();
+        });
+
+        it("should be able to unregister custom handle", function() {
+            dragHandleRegistrar.registerCustomHandle(customHandleEl);
+            dragHandleRegistrar.unregisterCustomHandle();
+
+            expect(dragHandleRegistrar.customHandle).toBeUndefined();
```

Comment:
You're missing a couple `expect()` (see comment on `ClrDragHandleRegistrar`):
- Listeners should be gone.
- CSS class should be gone.

Reply:
Added in the corresponding `expect()`s.